### DISCUSSION TITLE
Unable to render mesh [edit for posterity: check your imports]

### DISCUSSION THREAD

**Discussion by Polyterative** [nartc/angular-three#88] [Q&A] [answered]
I'm unable to render any kind of mesh for some reason. Scene is just blank.

Only `ngt-grid-helper` and   `ngt-soba-stars` works.

My code 
```html
<ngt-canvas
    (created)="$event.camera.lookAt(0, 0, 0)"
    [shadows]="true"
    [camera]="{ fov: 35, near: 0.25, far: 1000, position: [-25, 25, 25] }"
    [scene]="{ background: '#0e0e0e' | color, fog: ['#e0e0e0', 20, 100] | fog }"
    [size]="{ width: window.innerWidth, height: window.innerHeight }"
>
  <ngt-stats></ngt-stats>
  <ngt-soba-orbit-controls></ngt-soba-orbit-controls>
  <ngt-soba-stars></ngt-soba-stars>
  <ngt-ambient-light [color]="'#ffffff' | color"></ngt-ambient-light>
  <ngt-grid-helper
      #ngtGridHelper="ngtGridHelper"
      (ready)="onGridHelperReady(ngtGridHelper.helper)"
      [args]="[100, 40, '#000000', '#000000']"
      [receiveShadow]="true"
  ></ngt-grid-helper>
  <ngt-mesh [position]="[0,0,0]">
    <ngt-box-geometry></ngt-box-geometry>
    <ngt-mesh-physical-material [parameters]="{ color: '#ff0000' | color }"></ngt-mesh-physical-material>
  </ngt-mesh>

</ngt-canvas>
<!--<ngt-soba-loader></ngt-soba-loader>-->

```

Result
![image](https://user-images.githubusercontent.com/16295552/163468002-e8f10b9e-5a18-4141-8c3f-080cbd3ad590.png)

Reopenable project: https://github.com/Polyterative/AngularReactiveP5/blob/threejs/src/app/components/generative/container/container.component.html

**Comment by IRobot1** [comment, marked as answer]
Quick answer is that you're missing import of NgtAmbientLightModule, in generative.module.ts.  Once added, the ambient light you added to the scene has the desired effect.

![image](https://user-images.githubusercontent.com/25032599/163561915-0a548719-4f87-4264-80c6-675a9ed35ef1.png)

Because ng3 is very modular (to keep your app size small), its easy to miss an import.  I recommend organizing your imports by ng3 feature so its easy to keep track of the features you are pulling in...

```typescript
  imports: [
    CommonModule,

    NgtCoreModule,
    NgtStatsModule,
    NgtColorPipeModule,
    NgtFogPipeModule,

    NgtMeshModule,

    NgtObjectInputsControllerModule,

    NgtBoxGeometryModule,
    NgtConeGeometryModule,

    NgtAmbientLightModule,
    NgtHemisphereLightModule,
    NgtDirectionalLightModule,

    NgtBoxHelperModule,
    NgtGridHelperModule,

    NgtMeshBasicMaterialModule,
    NgtMeshPhongMaterialModule,
    NgtLineBasicMaterialModule,
    NgtMeshPhysicalMaterialModule,
    NgtMeshStandardMaterialModule,

    NgtSobaStarsModule,

    NgtSobaOrbitControlsModule,

    ThreeObjectsModule,
    SimpleCubeComponentModule,
  ]
```

**Reply by nartc** [reply]
Thanks @IRobot1. 
@Polyterative I'm working on a refactor that would help tremendously with IDEs that support auto importing modules for Angular (WebStorm). Not sure if VSCode will ever support it (or Angular Language Service for that matter) but overall, it will be easier because you'll get an error from compilation for missing a Module import instead of getting errors at runtime (or not even errors).

**Reply by IRobot1** [reply]
BTW, mixing P5 with ng3/3js is a very cool idea.

**Reply by Polyterative** [reply]
> Thanks @IRobot1. @Polyterative I'm working on a refactor that would help tremendously with IDEs that support auto importing modules for Angular (WebStorm). Not sure if VSCode will ever support it (or Angular Language Service for that matter) but overall, it will be easier because you'll get an error from compilation for missing a Module import instead of getting errors at runtime (or not even errors).

Thats great news since I totally rely on webstorm for auto imports :)
I will try importing it later today. Thanks for now!

This library is hyper cool, I'm super curious about what I will be able to build with it.

**Reply by Polyterative** [reply]
Ok that worked, I will be more mindful about imports from now on. Maybe improving the documentation would help for future developers
